mantidproject/mantid
# Expose Data Processor widget to python

Exposes the `DataProcessor` widget to Python and adds an example interface illustrating how to insert the widget in a Python interface. When reviewers are happy with this PR I will remove the python interface from the menu (but will keep the code) and resolve merge conflicts.

**To test:**

- It would be good if someone familiar with the `DataProcessor` UI reviewed code changes. I had to rewrite some code as there were methods (mainly those involving unique pointers) which I didn't know how to export. In addition, some methods were rewritten to return/take `QString`s instead of `std::string`s and `QSet` instead of `std::set` (otherwise those should be exported to Python too).
- It would be good if someone familiar with the reflectometry GUI tested that functionality has not changed:
  - All actions should work as before, both from the menus and from the toolbar.
  - Error/warning messages should be shown as before (for instance warnings when processing only one row in a group, etc).
  - Runs can be transferred from the search panel (for `OFFSPEC` try with the usual `1520249`, for `INTER` try with `1620340`).
  - The interface should still be able to reduce different types of runs (i.e. single-period and multi-period).
  - Any other functionality related to the `Runs` tab you can think of.
- Test the python interface (`Interfaces` > `Utility` > `DataProcessorInterface`).

Fixes #14732.

Release notes: internal change, no need to mention this in the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
